internal/operator/controller/contour/controller.go

```diff
@@ -311,7 +311,8 @@ func (r *reconciler) ensureContourDeleted(ctx context.Context, contour *operator
 	cli := r.client
 	if !contour.GatewayClassSet() {
 		if contour.Spec.NetworkPublishing.Envoy.Type == operatorv1alpha1.LoadBalancerServicePublishingType ||
-			contour.Spec.NetworkPublishing.Envoy.Type == operatorv1alpha1.NodePortServicePublishingType {
+			contour.Spec.NetworkPublishing.Envoy.Type == operatorv1alpha1.NodePortServicePublishingType ||
```

Comment:
Actually this condition may not be necessary as all Envoy types are covered.  

```
                if contour.Spec.NetworkPublishing.Envoy.Type == operatorv1alpha1.LoadBalancerServicePublishingType ||
                        contour.Spec.NetworkPublishing.Envoy.Type == operatorv1alpha1.NodePortServicePublishingType ||
                        contour.Spec.NetworkPublishing.Envoy.Type == operatorv1alpha1.ClusterIPServicePublishingType {
```

If I need to delete it, I will update it with this code and around L#261.